### ISSUE TITLE
Shape mismatch in dataloader when rasterization is not used.

### ISSUE THREAD

**Issue by LucaMarconato** [scverse/spatialdata#854]
Originally observed by @ilia-kats

When constructing a `ImageTileDataset` without using rasterization, a shape mismatch bug could occur (the returned tiles are not all of of the same size). This bug is for instance triggered when trying to construct a dataloader, as the collate function would fail.

Here below is the code to reproduce, on one of the dataset from https://github.com/giovp/spatialdata-sandbox. Uncommenting the 2 lines about rasterization show that the bug only affects the query in the intrinsic coordinate system (="pixel space").

```python
import spatialdata as sd
import torch

sdata = sd.read_zarr("/Users/macbook/embl/projects/basel/spatialdata-sandbox/visium_2.1.0_1_io/data.zarr")
IMAGE_ELEMENT = "CytAssist_FFPE_Human_Colon_Post_Xenium_Rep1_hires_image"
SHAPES_ELEMENT = "CytAssist_FFPE_Human_Colon_Post_Xenium_Rep1"
COORDINATE_SYSTEM = "downscaled_hires"

# sdata = sd.read_zarr("/Users/macbook/Desktop/mousebrain.zarr")
# IMAGE_ELEMENT = "mousebrain_hires_image"
# SHAPES_ELEMENT = "mousebrain"
# COORDINATE_SYSTEM = "downscaled_hires"

dataset = sd.dataloader.ImageTilesDataset(
    sdata,
    {SHAPES_ELEMENT: IMAGE_ELEMENT},
    {SHAPES_ELEMENT: COORDINATE_SYSTEM},
    tile_scale=1.5,
    # rasterize=True,
    # rasterize_kwargs={"target_width": 224},
)

dloader = torch.utils.data.DataLoader(
    [tile.images[IMAGE_ELEMENT].to_numpy() for tile in dataset],
    batch_size=sdata.shapes[SHAPES_ELEMENT].shape[0],
)

data = next(iter(dloader))
print(data.shape)
```

**Comment by LucaMarconato** [comment]
> **This whole message is AI-generated.** The issue was automatically discovered and reported by an AI agent (Claude) during an autonomous bug hunt on the `spatialdata` code base. It has **not** been verified or triaged by a human yet; the `needs: triage` label is set so that a maintainer can confirm it. The reproduction script below was executed by the agent in an isolated environment (see *Environment*) and its output is pasted verbatim.

I re-attempted to reproduce this issue on the current `main` branch (commit `ccf1ea048d054b6624214bf618008a9f9ae223e0`) as part of a triage pass over open issues, and confirmed it reproduces.

**Reproduction script** (PEP 723 inline metadata; run with `uv run repro.py`, no other setup needed):

```python
# /// script
# requires-python = ">=3.12"
# dependencies = [
#     "spatialdata @ git+https://github.com/scverse/spatialdata.git@ccf1ea048d054b6624214bf618008a9f9ae223e0",
#     "torch",
# ]
# ///
"""#854: ImageTilesDataset without rasterization returns tiles of different pixel sizes (collate fails)."""
import warnings
import numpy as np
import geopandas as gpd
from shapely.geometry import Point
from spatialdata import SpatialData
from spatialdata.dataloader import ImageTilesDataset
from spatialdata.models import Image2DModel, ShapesModel

warnings.simplefilter("ignore")
rng = np.random.default_rng(0)
img = Image2DModel.parse(rng.integers(0, 255, (1, 512, 512), dtype=np.uint8))
centres = rng.uniform(50, 450, (40, 2))  # non-integer centres
circles = ShapesModel.parse(gpd.GeoDataFrame({"geometry": [Point(*c) for c in centres], "radius": np.full(40, 10.3)}))
sdata = SpatialData(images={"img": img}, shapes={"cells": circles})
ds = ImageTilesDataset(sdata, regions_to_images={"cells": "img"}, regions_to_coordinate_systems={"cells": "global"}, tile_scale=1.5)
shapes = {tuple(ds[i].images["img"].shape) for i in range(len(ds))}
print("distinct tile shapes returned (rasterize=False):", sorted(shapes))
bug = len(shapes) > 1
print("VERDICT:", "REPRODUCED (tiles of different sizes; a DataLoader collate would fail)" if bug else "NOT REPRODUCED")
```

**Environment:** `uv run --no-project --python 3.13`, which resolves `spatialdata` from `git+https://github.com/scverse/spatialdata.git@ccf1ea048d054b6624214bf618008a9f9ae223e0` (extra dependencies, if any, are pinned in the script's inline metadata above).

**Output** (verbatim, only `uv` package-resolution/install log lines removed):

```
distinct tile shapes returned (rasterize=False): [(1, 30, 31), (1, 31, 30), (1, 31, 31)]
VERDICT: REPRODUCED (tiles of different sizes; a DataLoader collate would fail)
exit=0
```

**Agent's severity assessment:** medium (see the `priority: *` label added to this issue).

---
_Automatically generated; discovered by an AI agent (Claude) and not yet reviewed by a human._
